src/fsharp/TypeChecker.fs

```diff
@@ -13702,7 +13702,7 @@ module EstablishTypeDefinitionCores = begin
             if isForcedSuppressRelocate && canAccessFromEverywhere tycon.Accessibility && not cenv.isScript then 
                 errorR(Error(FSComp.SR.tcGeneratedTypesShouldBeInternalOrPrivate(),tcref.Range))
 
-            let isSuppressRelocate = cenv.g.isInteractive || isForcedSuppressRelocate
+            let isSuppressRelocate = (*cenv.g.isInteractive ||*) isForcedSuppressRelocate
```

Comment:
Just remove ``isSuppressRelocate`` and replace its uses by ``isForcedSuppressRelocate``